examples/p4est_2d_dgsem/elixir_euler_forward_step.jl

```diff
@@ -0,0 +1,155 @@
+  # set the freestream flow parameters
+  rho_freestream = 1.4
+  v1 = 3.0
+  v2 = 0.0
+  p_freestream = 1.0
+
+  prim = SVector(rho_freestream, v1, v2, p_freestream)
+  return prim2cons(prim, equations)
+end
+
+initial_condition = initial_condition_mach3_flow
+
+
+boundary_condition_inflow = BoundaryConditionDirichlet(initial_condition_mach3_flow)
+
+# Outflow boundary condition.
+# FIXME: For now only works for supersonic outflow where all values come from the internal state.
+# The bones are here for the subsonic outflow as well. One simply needs to pass the reference pressure
+# to set it from the outside, the rest comes from the internal solution.
+# Once fixed this function should probably move to `compressible_euler_2d.jl`
+# See the reference below for a discussion on inflow/outflow boundary conditions.
+#
+# - Jan-Reneé Carlson (2011)
+#   Inflow/Outflow Boundary Conditions with Application to FUN3D.
+#   [NASA TM 20110022658](https://ntrs.nasa.gov/citations/20110022658)
+@inline function boundary_condition_outflow(u_inner, normal_direction::AbstractVector, x, t,
+                                            surface_flux_function, equations::CompressibleEulerEquations2D)
+  # # This would be for the general case where we need to check the magnitude of the local Mach number
+  # norm_ = norm(normal_direction)
+  # # Normalize the vector without using `normalize` since we need to multiply by the `norm_` later
+  # normal = normal_direction / norm_
+
+  # # Rotate the internal solution state
+  # u_local = Trixi.rotate_to_x(u_inner, normal, equations)
+
+  # # Compute the primitive variables
+  # rho_local, v_normal, v_tangent, p_local = cons2prim(u_local, equations)
+
+  # # Compute local Mach number
+  # a_local = sqrt( equations.gamma * p_local / rho_local )
+  # Mach_local = abs( v_normal / a_local )
+  # if Mach_local <= 1.0
+  #   p_local = # Set to the external reference pressure value (somehow? maybe stored in `equations`)
+  # end
+
+  # # Create the `u_surface` solution state where the local pressure is possibly set from an external value
+  # prim = SVector(rho_local, v_normal, v_tangent, p_local)
+  # u_boundary = prim2cons(prim, equations)
+  # u_surface = Trixi.rotate_from_x(u_boundary, normal, equations)
+
+  # NOTE: Only for the supersonic outflow is this strategy valid
+  # Calculate the boundary flux entirely from the internal solution state
+  flux = Trixi.flux(u_inner, normal_direction, equations)
+
+  return flux
+end
+
+boundary_conditions = Dict( :Bottom     => boundary_condition_slip_wall,
+                            :Step_Front => boundary_condition_slip_wall,
+                            :Step_Top   => boundary_condition_slip_wall,
+                            :Top        => boundary_condition_slip_wall,
+                            :Right      => boundary_condition_outflow,
+                            :Left       => boundary_condition_inflow   )
+
+volume_flux = flux_shima_etal
```

Comment:
@gregorgassner said something about entropy, didn't he? Does it make a difference if you use an EC flux here?

Reply:
I will try it to see if it helps with the wall strangeness

Reply:
Done. It did help to switch to the entropy flux to remove this boundary layer (in addition to the adding local refinement near the corner)